target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-whw03v2.dts

```diff
@@ -0,0 +1,555 @@
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+    model = "Linksys WHW03 V2 (Velop)";
+    compatible = "linksys,whw03v2", "qcom,ipq4019";
+
+    aliases {
+        led-boot = &led_blue;
+        led-failsafe = &led_red;
+        led-running = &led_green;
+        led-upgrade = &led_red;
+    };
+
+    chosen {
+        bootargs-append = " root=/dev/ubiblock0_0 rootfstype=squashfs ro";
+        stdout-path = &blsp1_uart1;
+    };
+
+    memory {
+        device_type = "memory";
+        reg = <0x80000000 0x10000000>;
+    };
+
+    soc {
+        ess_tcsr@1953000 {
+            compatible = "qcom,tcsr";
+            reg = <0x1953000 0x1000>;
+            qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+        };
+
+
+        tcsr@1949000 {
+            compatible = "qcom,tcsr";
+            reg = <0x1949000 0x100>;
+            qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+        };
+
+        tcsr@194b000 {
+            compatible = "qcom,tcsr";
+            reg = <0x194b000 0x100>;
+            qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+        };
+
+        tcsr@1957000 {
+            compatible = "qcom,tcsr";
+            reg = <0x1957000 0x100>;
+            qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+        };
+    };
+
+
+    keys {
+        compatible = "gpio-keys";
+
+        reset {
+            label = "reset";
+            gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+            linux,code = <KEY_RESTART>;
+        };
+    };
+};
+
+
+&tlmm {
+    mdio_pins: mdio_pinmux {
+        mux_1 {
+            pins = "gpio6";
+            function = "mdio";  
+            bias-pull-up;
+        };
+
+        mux_2 {
+            pins = "gpio7";
+            function = "mdc";
+            bias-pull-up;
+        };
+    };
+
+    i2c_0_pins: i2c_0_pinmux {
+        mux {
+            function = "blsp_i2c0";
+            pins = "gpio20", "gpio21";
+            bias-disable;
+        };
+    };
+
+    serial_0_pins: serial0_pinmux {
+        mux {
+            pins = "gpio16", "gpio17";
+            function = "blsp_uart0";
+            bias-disable;
+        };
+    };
+
+    serial_1_pins: serial1_pinmux {
+        mux {
+            pins = "gpio8", "gpio9", "gpio10", "gpio11";
+            function = "blsp_uart1";
+            bias-disable;
+        };
+    };
+
+    spi_0_pins: spi_0_pinmux {
+        pinmux {
+            function = "blsp_spi0";
+            pins = "gpio13", "gpio14", "gpio15";
+        };
+
+        pinmux_cs {
+            function = "gpio";
+			pins = "gpio12";
+        };
+
+		pinconf {
+			pins = "gpio13", "gpio14", "gpio15";
+			drive-strength = <12>;
+			bias-disable;
+		};
+
+        pinconf_cs {
+            pins = "gpio12";
+            drive-strength = <2>;
+            bias-disable;
+            output-high;
+        };
+    };
+    
+    spi_1_pins: spi_1_pinmux {
```

Comment:
This whole node is a mess, all pins that are sharing the same properties go under one node.